APCAppCore/APCAppCore/DataSubstrate/Model/APCUser+Bridge.m

```diff
@@ -404,8 +414,12 @@ - (void)retrieveConsentOnCompletion:(void (^)(NSError *))completionBlock
                     }
                 });
             } else {
-                self.consentSignatureName = name;
-                self.consentSignatureImage = UIImagePNGRepresentation(signatureImage);
+                // parse consent signature dictionary, if we have one
+                if ([consentSignature isKindOfClass:[SBBConsentSignature class]]) {
+                    SBBConsentSignature *cSig = consentSignature;
+                    self.consentSignatureName = cSig.name;
+                    self.consentSignatureImage = UIImagePNGRepresentation([cSig signatureImage]);
```

Comment:
Any idea why this stores the PNG representation rather than the UIImage? I can't think why they'd want to do that, but if it's really necessary, it'd probably be cleaner to get it from the original image data in the ConsentSignature object rather than doing the roundtrip from base64 encoded string to UIImage to NSData:

self.consentSignatureImage = cSig.imageData ? [[NSData alloc] initWithBase64EncodedString:cSig.imageData options:kNilOptions] : nil;